### ISSUE TITLE
support serde Serialize/Deserialize

### ISSUE THREAD

**Issue by pmarks** [kennytm/extprim#10]
See https://github.com/brson/rust-api-guidelines#c-serde.  It seems like stabilization of u128 is still a ways off (https://github.com/rust-lang/rust/issues/35118).  Would there be interest in a PR to add serde support as a cfg feature? I'm willing to take a crack at it.

**Comment by kennytm** [comment]
Thanks, I saw #11. 

In #11 the Serialize and Deserialize are derived, so the 128-bit number will be serialized as something like this right?

```json
"number": {
    "lo": 123456,
    "hi": 654321
}
```

Would serializing it as a string better? Or is human-readability not worth the effort?

```json
"number": "12070092029053707523056192"
```

I've checked `num_bigint` and they just serialize as an array `"number": [123456, 654321]`. 

BTW I also noted that `serde` itself does not support native u128/i128, maybe you should also file an issue there.

**Comment by pmarks** [comment]
When using serde with a binary encoder (eg bincode) the u64-based representation will be much more efficient than encoding is as a string (this is my use case).  Trying to outsmart the auto-derived implementations sounds difficult & likely a maintenance burden.

Thanks for the note re the native types. I'll file a bug w/ serde.

**Comment by kennytm** [comment]
PR #11 has been merged and included in version 1.4.0 🙂.
